test/api_integration/apis/telemetry/utils/try_assertion_until.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+/**
+ * A helper function to retry an assertion multiple times before failing the test.
+ *
+ * @param subject async function to invoke for assertion.
+ * @param assertion assertion function; throw error or return false on fail assertion.
+ * @param numAttempts number of times to attempt before completely failing. (must be >= 1 - default 3)
+ * @param waitTimeMs time in ms to wait before attempting the next retry. (default none)
+ */
+export async function tryAssertionUntil<R>(
+  subject: () => Promise<R>,
+  assertion: (result: void | R, error?: Error) => false | void,
+  numAttempts: number = 3,
```

Comment:
Is it an equivalent of https://github.com/elastic/kibana/blob/111e15a0549df284fa3254057797a7359ffca9a8/test/common/services/retry/retry.ts#L39 ? the only difference your version specifies `numAttempts` instead of `timeout`. Can we use the standard library?